pallets/settlement/src/lib.rs

```diff
@@ -546,6 +596,33 @@ decl_module! {
             Ok(())
         }
 
+        /// Adds a new confidential instruction.
+        ///
+        /// # Arguments
+        /// * `venue_id` - ID of the venue this instruction belongs to.
+        /// * `settlement_type` - Defines if the instruction should be settled
+        ///    immediately after receiving all auths or waiting till a specific block.
```

Comment:
@ParnianAlimi @aafshar-poly I am not sure whether we need to support the `SettlementType::ExecuteOnAuthorization` or not for confidential transfers because it may cause problems related to compliance checks. Mediator has to validate the txn compliances on the basis of the future block instead of the current block at the time transaction justification.

Reply:
We'll work on transaction ordering after this one is merged. We can chat offline about this.